databricks/notebooks/rapids_airline_hyperopt.ipynb

```diff
@@ -0,0 +1,419 @@
+    "from hyperopt.pyll import scope\n",
+    "\n",
+    "search_space = [\n",
+    "        scope.int(hp.quniform('max_depth', 5, 15, 1)),\n",
+    "        hp.uniform('max_features', 0., 1.0),\n",
+    "        scope.int(hp.quniform('n_estimators', 100, 500, 100))\n",
+    "    ]\n",
+    "\n",
+    "algo = tpe.suggest\n",
+    "spark.conf.set('spark.task.maxFailures', '1')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Hyperopt with CPU"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark_trials = hyperopt.SparkTrials(parallelism=MAX_PARALLEL)\n",
+    "mlflow.end_run() # Close out any run in progress\n",
+    "\n",
+    "with mlflow.start_run() as run: \n",
+    "    mlflow.set_tag(\"mlflow.runName\", \"CPU_run\")\n",
+    "    best = fmin(\n",
+    "      fn=train_cpu,\n",
+    "      space=search_space,\n",
+    "      algo=algo,\n",
+    "      trials = spark_trials,\n",
+    "      max_evals=MAX_EVALS)\n",
+    "    mlflow.set_tag(\"best params\", str(best))\n",
+    "\n",
+    "    # Re-fit the best model on ALL of the data (no test set)\n",
+    "    print(best)\n",
+    "    train_cpu((int(best[\"max_depth\"]), best[\"max_features\"], int(best[\"n_estimators\"])),\n",
+    "            test_set_frac=0.0, registered_model_name=\"MLFlow_Airline_CPU_large_\")\n",
+    "\n",
+    "mlflow.end_run()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Hyperopt with GPU"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# SparkTrials object will automatically log runs to MLFlow in DataBricks\n",
+    "spark_trials = hyperopt.SparkTrials(parallelism=MAX_PARALLEL)\n",
+    "mlflow.end_run() # Close out any run in progress\n",
+    "\n",
+    "with mlflow.start_run() as run:\n",
+    "    mlflow.set_tag(\"mlflow.runName\", \"GPU_run\")\n",
+    "    best = fmin(fn=train_rapids,\n",
+    "      space=search_space,\n",
+    "      algo=algo,\n",
+    "      trials = spark_trials,\n",
+    "      max_evals=MAX_EVALS)\n",
+    "    mlflow.set_tag(\"best params\", str(best))\n",
+    "\n",
+    "    # Re-fit the best model on ALL of the data (no test set)\n",
+    "    train_rapids((int(best[\"max_depth\"]), best[\"max_features\"], int(best[\"n_estimators\"])),\n",
+    "               test_set_frac=0.0, registered_model_name=\"MLFlow_Airline_RAPIDS\")\n",
+    "mlflow.end_run()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Results\n",
+    "\n",
+    "On a `p3.2xlarge` cluster on Databricks, the runtimes observed were 24 minutes for GPU and nearly 14 hours for CPU. We illustrate that RAPIDS on GPU can give up to 35x speedups. This makes a huge difference when deciding whether or not to run a Hyperparamter optimzation.  "
```

Comment:
I love the numbers! Maybe we can rephrase the last sentence to: "Hopefully it will be easier to integrate hyperparameter optimization into your workflow if it can run in a coffee break rather than running overnight!"

(Also, you have a trailing blank cell at the end of the notebook that could be deleted)